src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -238,7 +272,91 @@ private void validateSsl() {
     }
   }
 
+  private void validateConnection() {
+    try {
+      if (!client.ping(RequestOptions.DEFAULT)) {
+        throw new IOException("Cannot connect to Elasticsearch.");
```

Comment:
Is it worth creating a relatively expensive expectation object(mostly due to stack trace allocation) and the catch it immediately? Maybe just factor it out in a separate method, e.g?
```
try{
..
if (!client.ping(RequestOptions.DEFAULT)) {
handleFailedPing("Failed to send a ping request to Elasticsearch");
} catch (IOException e) {
handleFailedPing(e.getMessage());
}
```